src/main/java/com/scalar/cassy/scheduler/CassyClient.java

```diff
@@ -153,40 +154,50 @@ private int awaitBackupStatusCompletedOrFailed(BackupResponse response, int time
         return 1;
       }
 
-      entry =
-          BackupListingResponse.newBuilder(
-                  blockingStub.listBackups(
-                      BackupListingRequest.newBuilder()
-                          .setClusterId(response.getClusterId())
-                          .setSnapshotId(response.getSnapshotId())
-                          .setTargetIp(response.getTargetIps(0))
-                          .setLimit(1)
-                          .build()))
-              .build()
-              .getEntries(0);
-
-      isNotCompletedOrFailed =
-          entry.getStatusValue() != OperationStatus.COMPLETED_VALUE
-              && entry.getStatusValue() != OperationStatus.FAILED_VALUE;
+      isNotCompletedOrFailed = false;
+      entries.clear();
+      for (int i = 0; i < response.getTargetIpsCount(); i++) {
+        entry =
+            BackupListingResponse.newBuilder(
+                blockingStub.listBackups(
+                    BackupListingRequest.newBuilder()
+                        .setClusterId(response.getClusterId())
+                        .setSnapshotId(response.getSnapshotId())
+                        .setTargetIp(response.getTargetIps(i))
+                        .setLimit(1)
+                        .build()))
+                .build()
+                .getEntries(0);
+        entries.add(entry);
+
+        if (entry.getStatusValue() != OperationStatus.COMPLETED_VALUE
+                && entry.getStatusValue() != OperationStatus.FAILED_VALUE) {
+          isNotCompletedOrFailed = true;
+        }
+      }
 
       if (isNotCompletedOrFailed) {
         Thread.sleep(2000); // so we aren't spamming Cassy server every millisecond
       }
     } while (isNotCompletedOrFailed);
 
+    StringBuilder ipStatus = new StringBuilder();
+    for (BackupListingResponse.Entry e : entries) {
+      ipStatus.append(String.format("IP %s: %s\n", e.getTargetIp(), e.getStatus().toString()));
+    }
+
     logger.info(
         String.format(
-            "\n\nThe following backup concluded with status %s\n\nCluster: %s\nSnapshot: %s\nTarget IP(s): %s\nBackup Type: %s\n",
-            entry.getStatus().toString(),
+            "\n\nThe %s backup for Cluster [%s] and Snapshot ID [%s] concluded as follows: \n\n%s\n",
+            BackupType.getByType(response.getBackupType()),
             response.getClusterId(),
             response.getSnapshotId(),
-            targetIps,
-            response.getBackupType()));
+            ipStatus.toString()));
 
-    if (entry.getStatusValue() != 3) {
-      return 1;
+    if (entries.stream().allMatch(e -> e.getStatusValue() == 3)) {
```

Comment:
Is there any particular reason of hard-coding `3`? Should it be `COMPLETED_VALUE`?